experimental/segmenter/src/word.rs

```diff
@@ -41,11 +41,28 @@ pub struct WordBreakIterator<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized>(
 
 derive_usize_iterator_with_type!(WordBreakIterator);
 
+/// The word type derived from the rule status in a rule-based word segmenter.
+#[non_exhaustive]
+#[derive(Copy, Clone, PartialEq, Debug)]
+#[repr(u8)]
+pub enum WordType {
+    /// No category tag
+    None = 0,
+    /// Number category tag
+    Number = 1,
+    /// Letter category tag, including CJK.
+    Letter = 2,
+}
+
 impl<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized> WordBreakIterator<'l, 's, Y> {
     /// Return the status value of break boundary.
```

Comment:
Suggestion: In #3281, I've changed this line. With you change, we should revise this. Maybe something like
"/// Return the word type tag value of the current word break boundary."

Reply:
How about (as suggested by @markusicu):

```
/// Returns the word type of the segment preceding the current boundary.
```

Reply:
This is better.